kimchi/src/verifier.rs

```diff
@@ -734,53 +719,43 @@ where
     //~~ - permutation commitment
     evaluations.push(Evaluation {
         commitment: proof.commitments.z_comm.clone(),
-        evaluations: proof.evals.iter().map(|e| e.z.clone()).collect(),
+        evaluations: proof.evals.iter().map(|e| vec![e.z]).collect(),
         degree_bound: None,
     });
 
     //~~ - index commitments that use the coefficients
-    evaluations.push(Evaluation {
-        commitment: index.generic_comm.clone(),
-        evaluations: proof
-            .evals
-            .iter()
-            .map(|e| e.generic_selector.clone())
-            .collect(),
-        degree_bound: None,
-    });
-    evaluations.push(Evaluation {
-        commitment: index.psm_comm.clone(),
-        evaluations: proof
-            .evals
-            .iter()
-            .map(|e| e.poseidon_selector.clone())
-            .collect(),
-        degree_bound: None,
-    });
+    if !index.generic_comm.is_empty() {
```

Comment:
Are you making this check because what you found out a few weeks ago that the PCS was complaining when you were passing [0]?

Reply:
yes! (see what I wrote on slack or in the description of this PR as well, it's a problem with zero commitments)